File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        int  s =0;
        int e = nums.size()-1;
        int ans =INT_MAX;
        if(nums[s] <= nums[e]) return nums[s];
        
        while(s<=e)
        {
            int mid  = s + (e-s)/2;
            if(nums[s] <= nums[mid])
            {
                ans = min(ans,nums[s]);
                s = mid+1;
            }
            else {
                ans = min(ans,nums[mid]);
                e = mid-1;
            }
        }
        
        return ans;
    }
};

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.java
class Solution {
    public int findMin(int[] nums) {
        if(nums.length == 1) return nums[0];
        if(nums[0] < nums[nums.length-1]) return nums[0];
        
        int s =0;
        int e = nums.length-1;
        int ans = Integer.MAX_VALUE;
        
        while(s<=e)
        {
            int mid = s + (e-s)/2;
            System.out.println(nums[mid]);
            if(nums[s] <= nums[mid])
            {
                ans = Math.min(ans,nums[s]);
                s= mid+1;
            }
            else {
                 ans = Math.min(ans,nums[mid]);
                e = mid-1;
            }
        }
        return ans;
    }
}